packages/core/src/storage/report.ts

```diff
@@ -42,10 +52,47 @@ interface StorageOperationContext {
   allowAppend: boolean;
 }
 
+interface WithHints {
+  title: string;
+  hints: string[];
+}
+
+function getExpectedGapSize(original: StorageField, updated: StorageField) {
+  const origEnd = storageFieldEnd(original);
+  const updatedStart = storageFieldBegin(updated);
+  const origNumBytes = original.type.item.numberOfBytes;
+  const origArraySize = original.type.tail;
+
+  if (
+    origEnd === undefined ||
+    updatedStart === undefined ||
+    origNumBytes === undefined ||
+    origArraySize === undefined
+  ) {
+    return undefined;
+  }
+
+  // use the floor since number of bytes may be larger than actual array (for non 32-byte types due to unused space after the array before the next slot)
+  const bytesPerItem = Math.floor(parseInt(origNumBytes) / parseInt(origArraySize, 10));
```

Comment:
BigInt division is integer division so it automatically applies floor. I removed the comment because it confused me more than it helped.

```suggestion
  const bytesPerItem = BigInt(origNumBytes) / BigInt(origArraySize);
```

Reply:
Changed it to `BigInt(parseInt(origTail, 10))` instead of `BigInt(origArraySize)` since the tail is a string like `49_storage`